impl/src/main/java/org/pentaho/di/trans/dataservice/ui/DataServiceTestResults.java

```diff
@@ -78,19 +78,35 @@ public void load( List<Object[]> rows, Listener tableKeyListener ) {
     container.layout();
   }
 
-  private TableView initTableView( List<Object[]> rows ) {
+  private TableView initTableView( int rowSize, Listener tableKeyListener ) {
     for ( Control control : container.getChildren() ) {
       control.dispose();
     }
     TableView tableView =
       new TableView( transMeta, container, SWT.NONE,
-        rowMetaToColumnInfo(), rows.size(), true, null,
+        rowMetaToColumnInfo(), rowSize, true, null,
         PropsUI.getInstance() );
     tableView.table.setItemCount( 0 );
-
+    tableView.table.addListener( SWT.KeyDown, tableKeyListener );
     return tableView;
   }
 
+  public void updateTableView( List<Object[]> rows ) {
```

Comment:
Is this method used anywhere else? Could we just cope with this one and discard the overloaded one below, and use the this.tableView inside it?

Reply:
discarded